### PR TITLE
Change the occourences of discordapp.com to discord.com 

### PR DESCRIPTION
**Pull request by Joshix-1**
see #554!
I changed everything I could find. I don't know if it is complete.

### CODE REVIEW

**Comment by Joshix-1** [comment]
I read somewhere, that the cdn.discordapp.com won't change.

**Comment by Bastian** [comment]
image
You missed one in the `mockserver.properties` file.

**Comment by Bastian** [comment]
> I read somewhere, that the cdn.discordapp.com won't change.

Ineed. Here's the reason:
image

**Comment by Bastian** [comment]
> Maybe a good solution will be to store the main url (discord.com) in a properties. load it on the begin of an app anreplace all real occuration (not javadoc) by this value

Agreed. Even though I think that it should be enough to have it as a `public static final String` constant somewhere:
```java
public static String DISCORD_DOMAIN = "discord.com";
public static String DISCORD_CDN_DOMAIN = "cnd.discordapp.com";
```

Maybe even in the API package in the `DiscordApi` interface?

**Comment by Vampire** [comment]
> You missed one in the mockserver.properties file.

Just have both in there.
It is for the generated SSL certificate for MockServer based tests.
Doesn't hurt to simply have both in there.

File: javacord-api/src/main/java/org/javacord/api/Javacord.java [hunk 1]
```diff
@@ -98,14 +98,14 @@
     /**
```
Comment (CharlesLgn) [inline javacord-api/src/main/java/org/javacord/api/Javacord.java:98]: you could put the two constant here

`public static final String URL_DISCORD = "discord.com"` ...

and then call this constants.

File: javacord-api/src/main/java/org/javacord/api/BotInviteBuilder.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
      * The base link of a bot invite.
      */
     public static final String BASE_LINK =
-            "https://discordapp.com/oauth2/authorize?client_id=%s&scope=bot&permissions=%s";
+            "https://discord.com/oauth2/authorize?client_id=%s&scope=bot&permissions=%s";
```
Comment (CharlesLgn) [inline javacord-api/src/main/java/org/javacord/api/BotInviteBuilder.java:15]: `"https://"+ DISCORD_URL +"/oauth2/authorize?client_id=%s&scope=bot&permissions=%s"`
or using `String.formatMessage`

File: javacord-api/src/main/java/org/javacord/api/entity/message/Message.java [hunk 3]
```diff
@@ -698,7 +698,7 @@ default String getReadableContent() {
      */
     default URL getLink() throws AssertionError {
         try {
-            return new URL("https://discordapp.com/channels/"
+            return new URL("https://discord.com/channels/"
```
Comment (CharlesLgn) [inline javacord-api/src/main/java/org/javacord/api/entity/message/Message.java:701]: `"https://"+ DISCORD_URL +"/channels/"`
or using `String.formatMessage`

File: javacord-api/src/main/java/org/javacord/api/util/DiscordRegexPattern.java [hunk 4]
```diff
@@ -78,18 +78,18 @@
...
-                            + "\\Qdiscordapp.com/channels/\\E # 'discordapp.com/channels/' \n"
-                            + "(?:(?<server>[0-9]++)|@me)     # '@me' or the server id as named group \n"
-                            + "/                              # '/' \n"
-                            + "(?<channel>[0-9]++)            # the textchannel id as named group \n"
-                            + "/                              # '/' \n"
-                            + "(?<message>[0-9]++)            # the message id as named group \n");
+            Pattern.compile("(?x)                              # enable comment mode \n"
+                            + "(?i)                            # ignore case \n"
+                            + "(?:https?+://)?+                # 'https://' or 'http://' or '' \n"
+                            + "discord(?:app)?\\.com/channels/ # 'discord(app).com/channels/' \n"
```
Comment (CharlesLgn) [inline javacord-api/src/main/java/org/javacord/api/util/DiscordRegexPattern.java:87]: why adding `(?:app)?\\` the url will only be `discord.com/channels/` no ?

File: javacord-core/src/main/java/org/javacord/core/util/rest/RestEndpoint.java [hunk 5]
```diff
@@ -153,7 +153,7 @@ public void setGlobal(boolean global) {
      * @return The full url of the endpoint.
      */
     public String getFullUrl(String... parameters) {
-        StringBuilder url = new StringBuilder("https://discordapp.com/api/v" + Javacord.DISCORD_API_VERSION + getEndpointUrl());
+        StringBuilder url = new StringBuilder("https://discord.com/api/v" + Javacord.DISCORD_API_VERSION + getEndpointUrl());
```
Comment (CharlesLgn) [inline javacord-core/src/main/java/org/javacord/core/util/rest/RestEndpoint.java:156]: same remark as other file (maybe have a constant for `https://discord.com`

File: javacord-api/src/main/java/org/javacord/api/util/DiscordRegexPattern.java (hunk 4 above)
Comment (Joshix-1) [reply javacord-api/src/main/java/org/javacord/api/util/DiscordRegexPattern.java:87]: Currently both is possible. And I would remove the (?:app)? only when only discordapp.com/channels is possible.

Comment (Vampire) [reply javacord-api/src/main/java/org/javacord/api/util/DiscordRegexPattern.java:87]: I'd also keep it, someone could post an old link. But add `+` after `?`, there is no need to backtrack.

Comment (Vampire) [reply javacord-api/src/main/java/org/javacord/api/util/DiscordRegexPattern.java]: And now the alignment again. :-)

**Comment by Bastian** [comment]
@Joshix-1 Do you plan to add the constants in this PR?

I would be ok if you don't. In this case I would create an issue for it and someone else (maybe @CharlesLgn ;-) ) can create a separate PR for it.

File: javacord-api/src/main/java/org/javacord/api/DiscordApi.java [hunk 6]
```diff
@@ -49,6 +49,8 @@
  * This class is the most important class for your bot, containing all important methods, like registering listener.
  */
 public interface DiscordApi extends GloballyAttachableListenerManager {
+    public static final String DISCORD_DOMAIN = "discord.com";
```
Comment (Bastian) [inline javacord-api/src/main/java/org/javacord/api/DiscordApi.java:52]: If it's in the public api, it should have Javadoc comments.
```suggestion

    /*
     * The domain of Discord.
     */
    public static final String DISCORD_DOMAIN = "discord.com";
```

File: javacord-api/src/main/java/org/javacord/api/DiscordApi.java [hunk 7]
```diff
@@ -49,6 +49,8 @@
...
+    public static final String DISCORD_CDN_DOMAIN = "cnd.discordapp.com";
```
Comment (Bastian) [inline javacord-api/src/main/java/org/javacord/api/DiscordApi.java:53]:
```suggestion
    /*
     * The domain of Discord's CDN.
     */
    public static final String DISCORD_CDN_DOMAIN = "cnd.discordapp.com";
```

File: javacord-api/src/main/java/org/javacord/api/BotInviteBuilder.java (hunk 2 above)
Comment (Bastian) [inline javacord-api/src/main/java/org/javacord/api/BotInviteBuilder.java:15]: When we have the constants, we should also use them.
```suggestion
            "https://" + DiscordApi.DISCORD_DOMAIN + "/oauth2/authorize?client_id=%s&scope=bot&permissions=%s";
```

File: javacord-api/src/main/java/org/javacord/api/entity/message/Message.java (hunk 3 above)
Comment (Bastian) [inline javacord-api/src/main/java/org/javacord/api/entity/message/Message.java:701]:
```suggestion
            return new URL("https://" + DiscordApi.DISCORD_DOMAIN + "/channels/"
```

File: javacord-api/src/main/java/org/javacord/api/DiscordApi.java (hunk 7 above)
Comment (DoNotSpamPls) [inline javacord-api/src/main/java/org/javacord/api/DiscordApi.java:53]: cnd != cdn

File: javacord-core/src/main/java/org/javacord/core/util/rest/RestEndpoint.java (hunk 5 above)
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/util/rest/RestEndpoint.java:156]:
```suggestion
        StringBuilder url = new StringBuilder("https://" + DiscordApi.DISCORD_DOMAIN + "/api/v" + Javacord.DISCORD_API_VERSION + getEndpointUrl());
```

**Pull request merged**